cmd/ef_tests/test_runner.rs

```diff
@@ -1,9 +1,34 @@
 use std::{collections::HashMap, path::Path};
 
-use ethereum_rust_core::evm::{execute_tx, SpecId};
-
 use crate::types::TestUnit;
+use ethereum_rust_core::{
+    evm::{execute_tx, SpecId},
+    rlp::{error::RLPDecodeError, structs::Decoder},
+    types::{BlockHeader, Transaction, Withdrawal},
+};
+use std::num::ParseIntError;
+
+pub fn decode_hex(s: &str) -> Result<Vec<u8>, ParseIntError> {
+    (0..s.len())
+        .step_by(2)
+        .map(|i| u8::from_str_radix(&s[i..i + 2], 16))
+        .collect()
+}
+
+/// Decodes a block and returns its header
+fn decode_block(rlp: &[u8]) -> Result<BlockHeader, RLPDecodeError> {
```

Comment:
Why do we have this function here? We should only use the decode from the core package

Reply:
I tried to use those decodes but I was getting an error because the blocks in the tests are encoded as 
```
[block_header, transactions, ommers, withdrawals]
```
however our `BlockHeader` and `BlockBody` decodes expect the encoding as
```
[block_header, [transactions, ommers, withdrawals]]
```
So I realized the easiest way to decode it was splitting the `BlockBody` into transactions, ommers, withdrawals.
What changes would be helpful here?

Reply:
I just added a `Block` struct along with its `RLPDecode` trait and all the necessary traits to convert `ef_tests::Block` to `core::Block`

Reply:
It seems the first one is the correct way to RLP encode an entire block. See formula 23 of the yellow paper: https://ethereum.github.io/yellowpaper/paper.pdf

Reply:
So we can add this https://github.com/lambdaclass/ethereum_rust/pull/183/files#diff-1ff4a558069929339178c0b3b1f36b64b738ab3aa10af261bb5ebe9f410aedaeR36-R40 and encode/decode it like `[block_header, transactions, ommers, withdrawals]`. So moving it out from `ef` and into `core`.

Reply:
Done!